context/Web3Context.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { ModalId, useModalContext } from "@/context/ModalContext";
+import { PropsWithChildren, createContext, useContext, useState } from "react";
+import { waitForTransactionReceipt } from "@wagmi/core";
+import { type WriteContractParameters, type Config } from "@wagmi/core";
+
+import { writeContract as wagmiWriteContract } from "wagmi/actions";
+import { TransactionModal } from "@/app/components";
+
+export interface Web3ContextProps {
+  writeContractShowModal: (
+    config: Config,
+    args: WriteContractParameters
+  ) => Promise<string | undefined>;
+}
+
+export const Web3Context = createContext<Web3ContextProps>({
+  writeContractShowModal: async (config, args) => "",
+});
+
+export const Web3Provider = ({ children }: PropsWithChildren) => {
```

Comment:
This Provider name is very generic. Let's call it `TransactionProvider`.

Reply:
Ok let's go with `TxProvider`